cmd/ipfs/init.go

```diff
@@ -142,6 +148,23 @@ func addTheWelcomeFile(conf *config.Config) error {
 	return nil
 }
 
+func initializeIpnsKeyspace(conf *config.Config) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	nd, err := core.NewIPFSNode(ctx, core.Offline(conf))
+	if err != nil {
+		return err
+	}
+	defer nd.Close()
```

Comment:
The DAG service needs the blockservice which is comprised of the blockstore and the exchange. These each have dependencies which need to be fulfilled. The wiring is non-trivial.

I'd be a bit worried about wiring up a complex object graph for a one-off operation. It's nice to be able to use a `core` as one simple building block. 

**a**
Reasons to use the core:
* less moving parts (externally)
* good to reuse/exercise the code paths (forcing function to make the process cheaper, faster, more reliable)

**b** 
Reasons to use raw components:
* Users may have a better experience plugging together the raw components themselves. Using the raw components whenever possible may drive us to improve their APIs.

Both a and b are probably important. It's unclear what'll be best for our users. First contact may reveal which style to (internally) encourage.